server/trillian_map_server.go

```diff
@@ -187,15 +187,26 @@ func (t *TrillianMapServer) SetLeaves(ctx context.Context, req *trillian.SetMapL
 	}
 
 	rootHash, err := smtWriter.CalculateRoot()
+	if err != nil {
+		return nil, fmt.Errorf("CalculateRoot(): %v", err)
+	}
 	newRoot := trillian.SignedMapRoot{
 		TimestampNanos: time.Now().UnixNano(),
 		RootHash:       rootHash,
 		MapId:          req.MapId,
 		MapRevision:    tx.WriteRevision(),
 		Metadata:       req.MapperData,
-		// TODO(al): Actually sign stuff, etc!
-		Signature: &spb.DigitallySigned{},
 	}
+	// Sign the root.
```

Comment:
(nit) Be nice to move this chunk up before `newRoot := ...` and then create a complete `SignedMapRoot` in one go with all relevant error paths already covered.